cli/configssh.go

```diff
@@ -244,7 +246,13 @@ func (r *RootCmd) configSSH() *serpent.Command {
 			sshConfigOpts.header = r.header
 			sshConfigOpts.headerCommand = r.headerCommand
 
-			recvWorkspaceConfigs := sshPrepareWorkspaceConfigs(inv.Context(), client)
+			// Talk to the API early to prevent bad placement of the
+			// version mismatch warning. The asynchronous requests
+			// issued by sshPrepareWorkspaceConfigs may trigger the
+			// warning at any time.
```

Comment:
You might want to clarify what "bad placement" means.

How will this change address the issue if
> The asynchronous requests issued by sshPrepareWorkspaceConfigs may trigger the warning at any time.

?

Reply:
Is this something we can test?

Reply:
Sure, I can clarify.

> How will this change address the issue if

By talking early. The warning is printed only once/on first API communication. Understanding this context requires knowledge of the version mismatch message, not sure it's worth expanding upon here. 🤔

> Is this something we can test?

Not really, since version mismatch is a http transport middleware triggered on first API comms, it applies essentially to _all_ CLI commands and testing this in a general way doesn't seem feasible or would require rewriting most of our tests to be run both with a matched and mismatched client/server.

Reply:
fb1d0d974

Reply:
Ah I see, thanks for the clarification; I think this detail would be relevant in the comment.
"Early" doesn't really explain why this fixes the issue, IMO.

Reply:
I do see your point Danny, and how that could be valuable information for the reader. I just don't think repeating the behavior about how version mismatch works in every place there needs to be an exception is the right way to go about it.

The curious reader can search for "version mismatch" and it/it's doc:

```
// wrapTransportWithVersionMismatchCheck adds a middleware to the HTTP transport
// that checks for version mismatches between the client and server. If a mismatch
// is detected, a warning is printed to the user.
```

I'm open to ideas for improvement, if you think it really needs it. I just want to avoid documenting another components logic.